src/sql/src/plan/statement/ddl.rs

```diff
@@ -681,18 +764,31 @@ pub fn plan_create_source(
                     );
                     column_casts.push(cast_expr);
                 }
-                table_casts.push(column_casts);
+                let r = table_casts.insert(i + 1, column_casts);
```

Comment:
Now that the above loop never fails we should be able to remove the `continue` and leave type here as `Vec<Vec<MirScalarExpr>>` since there is exactly one expr for each column for each table, right?

Reply:
Certainly possible, but requires more thoughtful accounting when retaining the requested table casts here. Glad to make that change if you prefer; was just trying to avoid as many off by one errors as possible.

```rust
    if let GenericSourceConnection::Postgres(PostgresSourceConnection { table_casts, .. }) =
        &mut external_connection
    {
        // Now that we know which subsources sources we want, we can remove all
        // unused table casts from this connection; this represents the
        // authoritative statement about which publication tables should be
        // used within storage.
        let used_pos: HashSet<_> = subsource_exports.values().collect();
        table_casts.retain(|pos, _| used_pos.contains(pos));
    }
 ```